homeassistant/components/venstar/sensor.py

```diff
@@ -155,9 +155,24 @@ def unique_id(self):
         return f"{self._config.entry_id}_{self.sensor_name.replace(' ', '_')}_{self.entity_description.key}"
 
     @property
-    def native_value(self) -> int:
+    def native_value(self) -> int | None:
         """Return state of the sensor."""
-        return self.entity_description.value_fn(self.coordinator, self.sensor_name)
+        try:
+            value = self.entity_description.value_fn(self.coordinator, self.sensor_name)
+            if value is None:
+                raise ValueError
+            return value
+        except (TypeError, ValueError):
+            self.coordinator.logger.debug(
+                "unable to set value for %s",
+                self.entity_description.value_fn(self.coordinator, self.sensor_name),
```

Comment:
Why are you calling here the function again?
If the function has raised a TypeError before, it will raise it here again so catching the TypeError is useless.